admin/src/react-components/service-editor.js

```diff
@@ -224,14 +225,45 @@ class ConfigurationEditor extends Component {
 
   renderLongTextInput(path, descriptor, currentValue) {
     const displayPath = path.join(" > ");
+
+    function isValidJSON(s) {
+      try {
+        JSON.parse(s);
+        return true;
+      } catch (error) {
+        console.error(error);
+        return false;
+      }
+    }
+
     return (
       <TextField
         key={displayPath}
         id={displayPath}
         label={descriptor.name || displayPath}
         inputProps={{ maxLength: 4096 }}
         value={currentValue || ""}
-        onChange={ev => this.onChange(path, ev.target.value)}
+        onChange={ev => {
+          if (descriptor.json) {
+            if (!isValidJSON(ev.target.value)) {
+              this.setState({
+                warningMessage: `Invalid JSON for ${descriptor.name || displayPath}. See console for details.`
+              });
+              console.error(`Invalid JSON for ${descriptor.name || displayPath}.`);
+              console.error(ev.target.value);
+            } else {
+              this.setState({ warningMessage: null });
```

Comment:
Can you add a check: `if (this.state.warningMessage !== null) this.setState({ warningMessage: null });` so we don't hammer this.setState unnecessarily.